jobbergate-api/jobbergate_api/email_notification.py

```diff
@@ -0,0 +1,76 @@
+"""
+Email notification system for Jobbergate.
+"""
+from dataclasses import dataclass
+from typing import List, Optional, Union
+
+from buzz import Buzz
+from loguru import logger
+from sendgrid import SendGridAPIClient  # type: ignore # no type hints or library stubs
+from sendgrid.helpers.mail import Mail  # type: ignore # no type hints or library stubs
+
+from jobbergate_api.config import settings
+
+
+class EmailNotificationError(Buzz):
+    """
+    Custom error to be raised for problems at the email notification system.
+    """
+
+    pass
+
+
+@dataclass
+class EmailManager:
+    """
+    Email manager.
+    """
+
+    email_client: SendGridAPIClient
+    from_email: Optional[str] = ""
+
+    def send_email(self, to_emails: Union[str, List[str]], subject: str, **kwargs) -> None:
+        """
+        Send an email using this manager.
+        """
+        message = self._build_message(to_emails, subject, **kwargs)
+
+        with EmailNotificationError.handle_errors("Error while sending email"):
+            self.email_client.send(message)
```

Comment:
I'm feeling we need more debug here, like, log the response from sendgrid. Also, I think if SendGrid is configured and the API cannot send the email, the API endpoint should return 500 or 502. 500 seems appropriate for me since we really do not know how to handle 😂

Reply:
Some of my initial tests concluded that there is no response from SendGrid if something goes wrong with `email_client.send()`. I'll investigate more because it would in fact help in logging and debugging.

Reply:
Just to confirm. When facing erros SendGrid raises:

* `python_http_client.exceptions.ForbiddenError: HTTP Error 403: Forbidden`
* `python_http_client.exceptions.UnauthorizedError: HTTP Error 401: Unauthorized`

Otherwise, the response has `status_code = 202` and an empty body.

Reply:
good, so I think we just log it as error and make the code move on

Reply:
we must make sure Sentry captures this otherwise we will not be aware of such a thing

Reply:
solved in [b97ad5a](https://github.com/omnivector-solutions/jobbergate/pull/119/commits/b97ad5ad4a784ea1e7102b3d5c662a780e189741).

Reply:
complemented by [7a03889](https://github.com/omnivector-solutions/jobbergate/pull/119/commits/7a038899fc1d11bd2674e1758da2a8e2ffec01c0).